sklearn/tree/_export.py

```diff
@@ -714,6 +723,21 @@ def recurse(self, node, tree, ax, max_x, max_y, depth=0):
                     (max_y - node.parent.y - 0.5) / max_y,
                 )
                 ax.annotate(node.tree.label, xy_parent, xy, **kwargs)
+
+                # Draw True/False labels if parent is root node
+                if node.parent.parent is None:
+                    # Adjust the position for the text to be slightly above the arrow
+                    text_pos = (
+                        (xy_parent[0] + xy[0]) / 2,
+                        (xy_parent[1] + xy[1]) / 2,
+                    )
+                    # Annotate the arrow with the edge label to indicate the child
+                    # where the sample-split condition is satisfied
+                    if node.parent.left() == node:
+                        label_text, label_ha = ("True  ", "right")
+                    else:
+                        label_text, label_ha = ("  False", "left")
```

Comment:
For other maintainers' reference, the spaces before and after the True/False label are to create some offset from the arrow on top of adjusting horizontal alignment. I'm not sure if there is a nicer-looking way to do this; we are also creating padding by using spaces in annotation text e.g. [this line](https://github.com/scikit-learn/scikit-learn/blob/fd28ffdc6f74e61c6d80a7c24986ef2101fb96cd/sklearn/tree/_export.py#L726).

Reply:
@adam2392 @Charlie-XIAO Idk is this intended but the arrows for the paths are almost not displayed when using `plot_tree()`, so the labels will be displayed behind the data boxes. Using `figsize=(20, 8)`, I just changed the `max_leaf_node` param a little, and it looks like this:
![decision_tree](https://github.com/scikit-learn/scikit-learn/assets/53251018/30186d24-5e2f-4b2b-ac21-05d4d7d6c32a)

Code for reproduction:
```python
from matplotlib import pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.datasets import load_iris
from sklearn.tree import DecisionTreeClassifier, plot_tree

iris = load_iris()
X = iris.data
y = iris.target

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)

clf = DecisionTreeClassifier(max_leaf_nodes=6, random_state=0)
clf.fit(X_train, y_train)

fig, ax = plt.subplots(figsize=(20, 8))
plot_tree(clf, feature_names=iris.feature_names,
          class_names=iris.target_names, filled=True, ax=ax)
```

Reply:
Thanks for pointing out @d-kleine. Yes I also noticed such cases, but IMHO users adjust their configurations (e.g. reduce font size) to make more spaces for the arrows and the labels. After all even the arrows themselves don't have enough space (let alone labels), and I don't think it is reasonable to display the labels above the nodes.

Reply:
Alright, thanks!

Reply:
@Charlie-XIAO Currently, it looks like displaying the labels in the tree is "hardcoded". What do you think about providing this feature of displaying the labels for the first split in a decision tree with a parameter for `plot_tree()` in case anyone does not like to display them? Maybe as a param called `display_labels` like in `[sklearn.metrics.ConfusionMatrixDisplay](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.ConfusionMatrixDisplay.html)`

Reply:
That seems reasonable to me, but I'll wait to see what the dev team thinks if it's adding complexity to the API.

Reply:
WDYT @thomasjpfan and @Charlie-XIAO ?

Reply:
For now, I'm okay with the current implementation. Adding `true` and `false` labels is already improvement.

Reply:
BTW resolves #17184